flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/DefaultExecutionGraphBuilder.java

```diff
@@ -329,7 +329,7 @@ public static DefaultExecutionGraph buildGraph(
     }
 
     public static boolean isCheckpointingEnabled(JobGraph jobGraph) {
-        return jobGraph.getCheckpointingSettings() != null;
+        return jobGraph.isCheckpointingEnabled();
```

Comment:
Let's revert this change and rename the method to `shouldInstantiateCheckpointCoordinator`.